brave/brave-core
# [ads] Add search result ad clicked InfoBar

**InfoBar on Desktop:**
<img width="1338" alt="image" src="https://github.com/user-attachments/assets/462255f9-bda6-4e57-8064-f21d2cb01a91">


**InfoBar on iOS:**
<img width="200" alt="image" src="https://github.com/user-attachments/assets/bbffb493-8bc0-4fa6-a39a-0b7abb54f375">

**InfoBar on Android:**
<img width="200" alt="image" src="https://github.com/user-attachments/assets/edad904c-09f3-489d-b752-0c159209517a">


Resolves https://github.com/brave/brave-browser/issues/39625



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

### The functionality should be tested for iOS, Android and Desktop

### Test case 1
- Fresh install
- Open staging search `search.brave.software` in a new tab
- Trigger search result ad for a staging server with the giraffe request
- Click search result ad
- Confirm that Search Result Ad Clicked InfoBar  was shown
- Click `Learn more`
- Make sure that Brave Search Result Ad help page is opened
- Open staging search `search.brave.software` in a new tab
- Trigger search result ad for a staging server with the giraffe request
- Click search result ad
- Confirm that Search Result Ad Clicked InfoBar was not shown for second time

### Test case 2
- Fresh install
- Open staging search `search.brave.software` in a new tab
- Trigger search result ad for a staging server with the giraffe request
- Click search result ad
- Confirm that Search Result Ad Clicked InfoBar  was shown
- Click a link in the page and navigate to a different page
- Confirm that the Infobar disappears automatically
- Open staging search `search.brave.software` in a new tab
- Trigger search result ad for a staging server with the giraffe request
- Click search result ad
- Confirm that Search Result Ad Clicked InfoBar was not shown for second time

### Test case 3
- Fresh install
- Open staging search `search.brave.software` in a new tab
- Trigger search result ad for a staging server with the giraffe request
- Click search result ad
- Confirm that Search Result Ad Clicked InfoBar  was shown
- Type a new URL in the URL bar and navigate to a page with entered URL
- Confirm that the Infobar disappears automatically
- Open staging search `search.brave.software` in a new tab
- Trigger search result ad for a staging server with the giraffe request
- Click search result ad
- Confirm that Search Result Ad Clicked InfoBar was not shown for second time
